boards: arm: black_f407ve: Add support for on-board SPI flash

bsteinsbo opened the pull request:
board: black_f407 Add support for on-board SPI flash Winbond W25Q16

erwango commented on boards/arm/black_f407_generic/black_f407ve_defconfig:
```diff
@@ -24,6 +24,9 @@ CONFIG_CLOCK_CONTROL=y
 CONFIG_GPIO=y
 
+# Enable SPI
+CONFIG_SPI=y
```
It is up to application to control the activation of peripherals (aside from basic ones: clocks, gpio, serial for console) 
This should be done on sample side (`samples/drivers/spi_flash/prj.cnf`)

bsteinsbo replied:
OK

erwango commented on samples/drivers/spi_flash/boards/black_f407ve.conf:
```diff
@@ -0,0 +1,8 @@
+#
```
Remove this line

bsteinsbo replied:
OK

erwango commented on samples/drivers/spi_flash/boards/black_f407ve.conf:
```diff
@@ -0,0 +1,8 @@
+#
+# Copyright (c) 2019 Peter Bigot Consulting, LLC
+# Copyright (c) 2020 Bjarne Steinsbo
+#
```
This line should be blank

bsteinsbo replied:
OK

erwango commented on samples/drivers/spi_flash/boards/black_f407ve.conf:
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2020 Bjarne Steinsbo
+#
+# SPDX-License-Identifier: Apache-2.0
+#
```
Remove this line

bsteinsbo replied:
OK.  And thanks for the review!

Gogo2309 committed:
boards: arm: black_f407ve: Add support for on-board SPI flash

This board contains an on-board Winbond W25Q16.
The chip is connected to spi1 using PB3-PB5 and PB0 as CS.

Signed-off-by: Bjarne Steinsbo <bjarne@gmail.com>
